WordPress/src/main/java/org/wordpress/android/ui/blaze/ui/blazeoverlay/BlazeOverlayFragment.kt

```diff
@@ -0,0 +1,282 @@
+import androidx.fragment.app.activityViewModels
+import coil.compose.rememberImagePainter
+import dagger.hilt.android.AndroidEntryPoint
+import org.wordpress.android.R
+import org.wordpress.android.ui.blaze.BlazeUiState
+import org.wordpress.android.ui.blaze.PostUIModel
+import org.wordpress.android.ui.compose.components.Drawable
+import org.wordpress.android.ui.compose.components.ImageButton
+import org.wordpress.android.ui.compose.theme.AppColor
+import org.wordpress.android.ui.compose.theme.AppTheme
+import org.wordpress.android.ui.compose.unit.FontSize
+import org.wordpress.android.ui.compose.unit.Margin
+import org.wordpress.android.ui.compose.utils.LocaleAwareComposable
+import org.wordpress.android.ui.utils.UiString
+import org.wordpress.android.util.LocaleManager
+
+@AndroidEntryPoint
+class BlazeOverlayFragment : Fragment() {
+    companion object {
+        fun newInstance() = BlazeOverlayFragment()
+    }
+
+    private val viewModel: BlazeViewModel by activityViewModels()
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View = ComposeView(requireContext()).apply {
+        setContent {
+            AppTheme {
+                val userLanguage by viewModel.refreshAppLanguage.observeAsState("")
+                val postModel by viewModel.promoteUiState.observeAsState(BlazeUiState.PromoteScreen.Site)
+                LocaleAwareComposable(
+                    locale = LocaleManager.languageLocale(userLanguage),
+                    onLocaleChange = viewModel::setAppLanguage
+                ) {
+                    viewModel.trackOverlayDisplayed()
+                    BlazeOverlayContent(postModel)
+                }
+            }
+        }
+    }
+
+    @Suppress("UNUSED_PARAMETER")
+    @Composable
+    fun BlazeOverlayContent(
+        postModelState: BlazeUiState.PromoteScreen,
+        isDarkTheme: Boolean = false
+    ) {
+        val post = when (postModelState) {
+            is BlazeUiState.PromoteScreen.PromotePost -> postModelState.postUIModel
+            else -> null
+        }
+        Column(
+            horizontalAlignment = Alignment.CenterHorizontally,
+            modifier = Modifier.padding(
+                top = Margin.ExtraLarge.value,
+                start = Margin.ExtraLarge.value,
+                end = Margin.ExtraLarge.value
+            )
+        ) {
+            Image(
+                painterResource(id = R.drawable.ic_blaze_promotional_image),
+                contentDescription = stringResource(id = R.string.blaze_activity_title),
+                modifier = Modifier.size(100.dp)
+            )
+            Text(
+                stringResource(id = R.string.blaze_promotional_text),
+                fontSize = FontSize.DoubleExtraLarge.value,
+                fontWeight = FontWeight.Bold,
+                modifier = Modifier.padding(top = Margin.ExtraExtraMediumLarge.value),
```

Comment:
np: I prefer to _always_ put `modifier`'s last, because they have a higher chance of being changed and you don't have to deal with the trailing comma anymore 😅

Reply:
Yeah that makes sense. 👍🏼. Thanks for that tip. I have made the change in 8e0a301af9248a9be3eb8a4f02d463e9c39dabfc